cos-fleetshard-operator-debezium/src/main/java/org/bf2/cos/fleetshard/operator/debezium/DebeziumOperandController.java

```diff
@@ -151,201 +134,392 @@ protected List<HasMetadata> doReify(
+        //
+
+        connectorDeployment.setSpec(new DeploymentSpec());
+        connectorDeployment.getSpec().setProgressDeadlineSeconds(60);
+        connectorDeployment.getSpec().setReplicas(1);
+        connectorDeployment.getSpec().setRevisionHistoryLimit(10);
+        connectorDeployment.getSpec().setSelector(selector(connector));
+        connectorDeployment.getSpec().setStrategy(new DeploymentStrategy(null, "Recreate"));
+
+        //
+        // Pod Template
+        //
+
+        connectorDeployment.getSpec().setTemplate(new PodTemplateSpecBuilder()
+            .withMetadata(
+                new ObjectMetaBuilder()
+                    .addToLabels(Resources.LABEL_CLUSTER_ID, connector.getSpec().getClusterId())
+                    .addToLabels(Resources.LABEL_CONNECTOR_ID, connector.getSpec().getConnectorId())
+                    .addToLabels(Resources.LABEL_DEPLOYMENT_ID, connector.getSpec().getDeploymentId())
+                    .build())
+            .withSpec(
+                new PodSpecBuilder()
+                    .withImagePullSecrets(configuration.imagePullSecretsName())
+                    .withVolumes(
+                        new VolumeBuilder()
+                            .withName("connector-data")
+                            .withPersistentVolumeClaim(
+                                new PersistentVolumeClaimVolumeSource(connectPvc.getMetadata().getName(), false))
+                            .build(),
+                        volume(connectSecret),
+                        volume(connectConfig))
+                    .withContainers(new ContainerBuilder()
+                        .withName("connector")
+                        .withTerminationMessagePath("/dev/termination-log")
+                        .withTerminationMessagePolicy("File")
+                        .withImage(shardMetadata.getContainerImage())
+                        .withImagePullPolicy("IfNotPresent")
+                        // triggers a re-deployment upon secret/config change
+                        .addToEnv(env("CONNECTOR_PARAMS_CHECKSUM", Secrets.computeChecksum(connectSecret)))
+                        .addToEnv(env("CONNECTOR_RESOURCES_CHECKSUM", ConfigMaps.computeChecksum(connectConfig)))
+                        .addToEnv(env("GC_LOG_ENABLED", "false"))
+                        .addToEnv(env("LOG_DIR", KAFKA_HOME))
+                        .addToEnv(env("GC_LOG_ENABLED", "false"))
+                        .addToEnv(env("KAFKA_GC_LOG_OPTS", " "))
+                        .addToEnv(env("KAFKA_LOG4J_OPTS", "-Dlog4j.configuration=file:" + LOGGING_CONFIG_PATH))
+                        // TODO: this is likely needed to be improved, ideally the jmx_prometheus_javaagent jar
+                        //       should be on a dedicated directory with a stable name
+                        .addToEnv(env("KAFKA_OPTS",
+                            "-javaagent:/opt/kafka/libs/jmx_prometheus_javaagent-0.17.2.redhat-00001.jar=9404:"
```

Comment:
This should be configurable from the shard metadata (at least the version part) where the container image is referenced because this version is image specific.